plutus-ledger/src/Ledger/Index.hs

```diff
@@ -397,11 +350,15 @@ minAdaTxOut = Ada.lovelaceOf 2_000_000
 -- slightly more expensive than 1 Ada. So, to be on the safe side, we set the
 -- minimum Ada of each transaction output to 2 Ada.
 checkMinAdaInTxOutputs :: ValidationMonad m => Tx -> m ()
-checkMinAdaInTxOutputs t@Tx { txOutputs } =
-    for_ txOutputs $ \txOut ->
-        if Ada.fromValue (txOutValue txOut) >= minAdaTxOut
+checkMinAdaInTxOutputs t@Tx { txOutputs } = do
+    params <- vctxParams <$> ask
+    for_ txOutputs $ \txOut -> do
+        let
+            minAdaTxOut' = either (const minAdaTxOut) id $
+                fromPlutusTxOutUnsafe params txOut <&> \txOut' -> Ada.fromValue $ evaluateMinLovelaceOutput params txOut'
```

Comment:
It seems we call `Ada.fromValue` after every `evaluateMinLovelaceOutput` call. Better fix `evaluateMinLovelaceOutput` to return `Ada`.